site/app/controllers/PollController.php

```diff
@@ -350,8 +360,18 @@ public function submitEdits(): RedirectResponse {
             if ($current_file_path !== null) {
                 unlink($current_file_path);
             }
-            $file_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "uploads", "polls", "poll_image_" . $_POST["poll_id"] . "_" . $_FILES["image_file"]["name"]);
-            move_uploaded_file($file["tmp_name"], $file_path);
+            $max_size = Utils::returnBytes(ini_get('upload_max_filesize'));
+            $acceptable = ['image/jpeg','image/jpg','image/gif','image/png'];
+            if (($_FILES['image_file']['size'] >= $max_size) || ($_FILES["image_file"]["size"] == 0)) {
+                $this->core->addErrorMessage("File(s) uploaded too large.  Maximum size is " . ($max_size / 1024) . " kb.");
+            }
+            elseif ((!in_array($_FILES['image_file']['type'], $acceptable)) && (!empty($_FILES["image_file"]["type"]))) {
+                $this->core->addErrorMessage("Upload failed: Invalid file type. Only JPG, GIF and PNG types are accepted.");
+            }
+            else {
+                $file_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "uploads", "polls", "poll_image_" . $_POST["poll_id"] . "_" . $_FILES["image_file"]["name"]);
+                move_uploaded_file($file["tmp_name"], $file_path);
```

Comment:
same comment here, use above validation strategy and propagate errors if any back to the user